### PR TITLE
Improve collision detection mechanism.

### PR DESCRIPTION
**Pull request by ioan994**
Reset only vertical/horizontal component speed vector depending on how object collides.

### CODE REVIEW

File: platformerLib/GameEngine.cpp [hunk 1]
```diff
@@ -46,7 +46,8 @@ void GameEngine::CheckGraySquareActorLocation(PhysicalObject* actor,
 
 	Distance new_x(actor->GetLocation().x.Value());
 	Distance new_y(actor->GetLocation().y.Value());
-	bool is_touched = false;
+   bool is_touched_horizontally = false;
```
Comment (vadim94) [inline platformerLib/GameEngine.cpp]: Incorrect indents here and below.
Can't be submitted to master :(

**Review by vadim94 — CHANGES_REQUESTED** [review]

File: platformerLib/GameEngine.cpp (hunk 1 above)
Comment (ioan994) [reply platformerLib/GameEngine.cpp]: We are going to use spaces, aren't we?

Comment (vadim94) [reply platformerLib/GameEngine.cpp]: Yes, but we shouldn't leave both spaces and tabs in the source file (it looks ugly).
Please, make the same style in the whole file.

**Comment by ioan994** [comment]
Updated formatting

**Pull request merged**